src/main/haskell/kore/test/Test/Kore/OnePath/Verification.hs

```diff
@@ -0,0 +1,349 @@
+module Test.Kore.OnePath.Verification
+    ( test_onePathVerification
+    ) where
+
+import Test.Tasty
+       ( TestTree )
+import Test.Tasty.HUnit
+       ( testCase )
+
+import           Control.Monad.Trans.Except
+                 ( runExceptT )
+import           Data.Default
+                 ( def )
+import qualified Data.Map as Map
+import           Data.Reflection
+                 ( give )
+import           Numeric.Natural
+                 ( Natural )
+
+import           Data.Limit
+                 ( Limit (..) )
+import           Kore.AST.MetaOrObject
+import           Kore.ASTUtils.SmartConstructors
+                 ( mkVar )
+import           Kore.IndexedModule.MetadataTools
+                 ( MetadataTools (..), SymbolOrAliasSorts )
+import qualified Kore.OnePath.Verification as OnePath
+import           Kore.Predicate.Predicate
+                 ( makeEqualsPredicate, makeNotPredicate, makeTruePredicate )
+import           Kore.Step.AxiomPatterns
+                 ( RewriteRule (RewriteRule), RulePattern (RulePattern) )
+import           Kore.Step.AxiomPatterns as RulePattern
+                 ( RulePattern (..) )
+import           Kore.Step.ExpandedPattern
+                 ( Predicated (Predicated) )
+import           Kore.Step.ExpandedPattern as Predicated
+                 ( Predicated (..) )
+import           Kore.Step.ExpandedPattern as ExpandedPattern
+                 ( CommonExpandedPattern, fromPurePattern )
+import           Kore.Step.Pattern
+                 ( CommonStepPattern )
+import           Kore.Step.Simplification.Data
+                 ( evalSimplifier )
+import qualified Kore.Step.Simplification.Simplifier as Simplifier
+import           Kore.Step.StepperAttributes
+import qualified SMT
+
+import           Test.Kore.Comparators ()
+import qualified Test.Kore.IndexedModule.MockMetadataTools as Mock
+import qualified Test.Kore.Step.MockSimplifiers as Mock
+import qualified Test.Kore.Step.MockSymbols as Mock
+import           Test.Tasty.HUnit.Extensions
+
+test_onePathVerification :: [TestTree]
+test_onePathVerification = give symbolOrAliasSorts
+    [ testCase "Runs zero steps" $ do
+        -- Axiom: a => b
+        -- Claim: a => b
+        -- Expected: error a
+        actual <- runVerification
+            metadataTools
+            (Limit 0)
+            [simpleAxiom Mock.a Mock.b]
+            [simpleClaim Mock.a Mock.b]
+        assertEqualWithExplanation ""
+            (Left $ ExpandedPattern.fromPurePattern Mock.a)
+            actual
+    , testCase "Runs one step" $ do
```

Comment:
I was confused here because I thought it should verify it in one step. Maybe add a comment about it